fix: Restore internals of strhash to compile on 32 bit architectures
https://github.com/AcademySoftwareFoundation/OpenImageIO/commit/004a04fc1d5f5a949598b18978f8f960fc604d1b

Index: src/include/OpenImageIO/detail/farmhash.h
--- src/include/OpenImageIO/detail/farmhash.h.orig
+++ src/include/OpenImageIO/detail/farmhash.h
@@ -2097,7 +2097,7 @@ STATIC_INLINE uint64_t Hash64(const char* s, size_t le
 // May change from time to time, may differ on different platforms, may differ
 // depending on NDEBUG.
 STATIC_INLINE size_t Hash(const char* s, size_t len) {
-  return sizeof(size_t) == 8 ? Hash64(s, len) : Hash32(s, len);
+  return sizeof(size_t) == 8 ? size_t(Hash64(s, len)) : size_t(Hash32(s, len));
 }
 
 // Hash function for a byte array.  For convenience, a 64-bit seed is also
